static/app/components/events/contexts/utils.tsx

```diff
@@ -63,6 +65,39 @@ export function getRelativeTimeFromEventDateCreated(
   );
 }
 
+// Typescript doesn't have types for DisplayNames yet and that's why the type assertion "any" is needed below.
+// There is currently an open PR that intends to introduce the types https://github.com/microsoft/TypeScript/pull/44022
+export function getFullLanguageDescription(locale: string) {
+  const sentryAppLanguageCode = ConfigStore.get('languageCode');
+
+  const [languageAbbreviation, countryAbbreviation] = locale.includes('_')
+    ? locale.split('_')
+    : locale.split('-');
+
+  try {
+    const languageNames = new (Intl as any).DisplayNames(sentryAppLanguageCode, {
+      type: 'language',
+    });
+
+    const languageName = languageNames.of(languageAbbreviation);
+
+    if (countryAbbreviation) {
+      const regionNames = new (Intl as any).DisplayNames(sentryAppLanguageCode, {
+        type: 'region',
+      });
+
+      const countryName = regionNames.of(countryAbbreviation.toUpperCase());
+
+      return `${languageName} (${countryName})`;
+    }
+
+    return languageName;
+  } catch (error) {
+    Sentry.captureException(error);
```

Comment:
I'm not familiar with how many browsers have `Intl.DisplayNames` should we check for it first before logging an exception

Reply:
nevermind seems okay https://caniuse.com/mdn-javascript_builtins_intl_displaynames_displaynames